tests/unit/classifier/GaussianProcessClassification_unittest.cc

```diff
@@ -67,68 +67,402 @@
+		// probit likelihood
+		CLogitVGLikelihood* likelihood = new CLogitVGLikelihood();
 
-	// choose Gaussian kernel with sigma = 2 and zero mean function
-	CGaussianKernel* kernel=new CGaussianKernel(10, 2);
-	CZeroMean* mean=new CZeroMean();
+		CKLCovarianceInferenceMethod* inf = new CKLCovarianceInferenceMethod(
+		    kernel, features_train, mean, labels_train, likelihood);
 
+		// train Gaussian process binary classifier
+		gpc = new CGaussianProcessClassification(inf);
+		gpc->train();
+	}
+	virtual void TearDown()
+	{
+		GaussianProcessClassification::TearDown();
+		SG_UNREF(gpc);
+	}
+	float64_t abs_tolerance;
+	const float64_t rel_tolerance = 1e-2;
+	CGaussianProcessClassification* gpc;
+};
+
+class GaussianProcessClassificationUsingKLCholesky
+    : public GaussianProcessClassification
+{
+public:
+	virtual void SetUp()
+	{
+		GaussianProcessClassification::SetUp();
+
+		// probit likelihood
+		CLogitVGLikelihood* likelihood = new CLogitVGLikelihood();
+
+		CKLCholeskyInferenceMethod* inf = new CKLCholeskyInferenceMethod(
+		    kernel, features_train, mean, labels_train, likelihood);
+
+		// train Gaussian process binary classifier
+		gpc = new CGaussianProcessClassification(inf);
+		gpc->train();
+	}
+	virtual void TearDown()
+	{
+		GaussianProcessClassification::TearDown();
+		SG_UNREF(gpc);
+	}
+	float64_t abs_tolerance;
+	const float64_t rel_tolerance = 1e-2;
+	CGaussianProcessClassification* gpc;
+};
+
+class GaussianProcessClassificationUsingKLDiagonal
+    : public GaussianProcessClassification
+{
+public:
+	virtual void SetUp()
+	{
+		GaussianProcessClassification::SetUp();
+
+		// probit likelihood
+		CLogitVGLikelihood* likelihood = new CLogitVGLikelihood();
+
+		CKLDiagonalInferenceMethod* inf = new CKLDiagonalInferenceMethod(
+		    kernel, features_train, mean, labels_train, likelihood);
+
+		// train Gaussian process binary classifier
+		gpc = new CGaussianProcessClassification(inf);
+		gpc->train();
+	}
+	virtual void TearDown()
+	{
+		GaussianProcessClassification::TearDown();
+		SG_UNREF(gpc);
+	}
+	float64_t abs_tolerance;
+	const float64_t rel_tolerance = 1e-2;
+	CGaussianProcessClassification* gpc;
+};
+
+class GaussianProcessClassificationUsingKLDual
+    : public GaussianProcessClassification
+{
+public:
+	virtual void SetUp()
+	{
+		GaussianProcessClassification::SetUp();
+
+		// probit likelihood
+		CLogitDVGLikelihood* likelihood = new CLogitDVGLikelihood();
+
+		CKLDualInferenceMethod* inf = new CKLDualInferenceMethod(
+		    kernel, features_train, mean, labels_train, likelihood);
+
+		// train Gaussian process binary classifier
+		gpc = new CGaussianProcessClassification(inf);
+		gpc->train();
+	}
+	virtual void TearDown()
+	{
+		GaussianProcessClassification::TearDown();
+
+		SG_UNREF(gpc);
+	}
+	float64_t abs_tolerance;
+	const float64_t rel_tolerance = 1e-2;
+	CGaussianProcessClassification* gpc;
+};
+
+// empty class since TEST and TEST_F cannot be used in the same file
```

Comment:
I am not sure I understand what you mean here. I definitely had `TEST` and `TEST_F` in the same file before ...
Also not very pretty :/

Reply:
You are right.. I was under the impression that TEST and TEST_F cannot be used in the same file. Should I remove the empty class and change the corresponding `TEST_F`s to `TEST`s?